extensions/data-workspace/src/services/workspaceService.ts

```diff
@@ -89,31 +89,36 @@ export class WorkspaceService implements IWorkspaceService {
 	 * @param projectType The file extension of the project, if not specified, all project provider extensions will be loaded.
 	 */
 	private async ensureProviderExtensionLoaded(projectType: string | undefined = undefined): Promise<void> {
-		const inactiveExtensions = vscode.extensions.all.filter(ext => !ext.isActive);
 		const projType = projectType ? projectType.toUpperCase() : undefined;
 		let extension: vscode.Extension<any>;
-		for (extension of inactiveExtensions) {
+		for (extension of vscode.extensions.all) {
 			const projectTypes = extension.packageJSON.contributes && extension.packageJSON.contributes.projects as string[];
 			// Process only when this extension is contributing project providers
 			if (projectTypes && projectTypes.length > 0) {
 				if (projType) {
 					if (projectTypes.findIndex((proj: string) => proj.toUpperCase() === projType) !== -1) {
-						await this.activateExtension(extension);
+						await this.handleProjectProviderExtension(extension);
 						break;
 					}
 				} else {
-					await this.activateExtension(extension);
+					await this.handleProjectProviderExtension(extension);
 				}
 			}
 		}
 	}
 
-	private async activateExtension(extension: vscode.Extension<any>): Promise<void> {
+	private async handleProjectProviderExtension(extension: vscode.Extension<any>): Promise<void> {
 		try {
-			await extension.activate();
+			if (!extension.isActive) {
+				await extension.activate();
+			}
 		} catch (err) {
 			Logger.error(ExtensionActivationErrorMessage(extension.id, err));
 		}
+
+		if (extension.isActive && extension.exports && !ProjectProviderRegistry.providers.includes(extension.exports)) {
```

Comment:
extension.exports is the IProjectProvider returned when activating the extension?

Reply:
yes, the object returned by the activate method is accessible by extension.exports

Reply:
(note that using .exports requires the extension be activated already. So you should either ensure that activate() has been called or that the calling extension has a dependency on the target extension or this may fail)

Reply:
yes, I am checking extension.isActive priority to extension.exports, also double checked that just like C#, for And operator, Javascript will skip evaluating the rest of the condition if the first part returns false.

Reply:
Yeah I wasn't saying what you were doing was wrong. Just providing more context for Kim or anyone else looking at this about what the difference is between calling activate() and using .exports directly.

Reply:
Although that does bring up a question - what if an extension contributes this but isn't active yet? Do you hook into extensions activating and make sure to update whenever new ones are activated?

Reply:
![image](https://user-images.githubusercontent.com/13777222/93401947-4c4a0d80-f838-11ea-8549-df5280f7dbb3.png)

just a couple lines above, I will try to activate the extension first